File: EcoSim.Core/Entities/Vegetation.cs
using System;

namespace EcoSim.Core.Entities
{
    /// <summary>
    /// Static plant that regrows and can be eaten.
    /// </summary>
    public class Vegetation : Entity, IEdible
    {
        private readonly int _maxHealth;
        private readonly Random _rng = new();

        public int FoodValue    { get; private set; }
        public int RegrowthRate { get; }

        public Vegetation(int initialHealth, int lifespan, int foodValue, int regrowthRate)
            : base(initialHealth, lifespan)
        {
            _maxHealth    = initialHealth;
            FoodValue     = foodValue;
            RegrowthRate  = regrowthRate;  // e.g. probability% to regrow 1hp each tick
        }

        public override void Update()
        {
            base.Update();
            if (IsDead) return;

            // passive regrowth
            if (Health < _maxHealth && _rng.NextDouble() < RegrowthRate / 100.0)
                Health++;
        }

        /// <summary>
        /// Called by animal when they eat this plant.
        /// </summary>
        public void BeEaten(int amount)
        {
            Health = Math.Max(0, Health - amount);
            if (Health == 0) IsDead = true;
        }
    }
}

File: entities/vegetation.py
from .base import Entity, Consumable
import random

class Vegetation(Entity, Consumable):
    def __init__(self, x, y, health=5, age=0, lifespan=100, food_value=3):
        super().__init__(x, y, health, age, lifespan)
        self._food_value = food_value

    @property
    def food_value(self):
        return self._food_value

    def tick(self, world):
        # Age and health changes
        self.age += 1
        if self.age >= self.lifespan:
            world.remove_entity(self)
        # Passive growth: occasional health regen
        if random.random() < 0.1:
            self.health = min(self.health + 1, 10)